sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/implementation/task/TaskManager.java

```diff
@@ -69,41 +73,45 @@ public Exception getException() {
-                        exception = (Exception) e;
-                        pendingList.addAll(taskList);
-                    }
-                }).subscribe(response -> {
-                    if (response != null && response.getValue() != null) {
-                        for (BatchTaskAddResult result : response.getValue()) {
-                            if (result.getError() != null) {
-                                if (result.getStatus() == BatchTaskAddStatus.SERVER_ERROR) {
-                                    // Server error will be retried
-                                    for (BatchTaskCreateContent batchTaskToCreate : taskList) {
-                                        if (batchTaskToCreate.getId().equals(result.getTaskId())) {
-                                            pendingList.add(batchTaskToCreate);
-                                            break;
-                                        }
+                BatchCreateTaskCollectionResult response
+                    = taskSubmitter.submitTasks(jobId, new BatchTaskGroup(taskList));
+
+                if (response != null && response.getValues() != null) {
+                    for (BatchTaskCreateResult result : response.getValues()) {
+                        if (result.getError() == null) {
+                            continue; // success
+                        }
+
+                        if (result.getStatus() == BatchTaskAddStatus.SERVER_ERROR) {
+                            // Requeue this single task for retry
+                            String id = result.getTaskId();
+                            if (id != null) {
+                                for (BatchTaskCreateParameters p : taskList) {
+                                    if (id.equals(p.getId())) {
+                                        pendingList.add(p);
+                                        break;
                                     }
-                                } else if (result.getStatus() == BatchTaskAddStatus.CLIENT_ERROR
-                                    && !result.getError().getMessage().getValue().contains("Status code 409")) {
-                                    // Client error will be recorded
-                                    failures.add(result);
                                 }
                             }
+                        } else if (result.getStatus() == BatchTaskAddStatus.CLIENT_ERROR) {
+                            BatchError err = result.getError();
+                            String code = (err != null) ? err.getCode() : null;
+                            if (!"TaskExists".equalsIgnoreCase(code)) {
+                                failures.add(result);
+                            }
                         }
                     }
-                });
+                }
+            } catch (HttpResponseException e) {
+                handleHttpException(e, taskList);
+            } catch (RuntimeException e) {
```

Comment:
I don't think this catch block is doing anything. It's covered by the `catch (Exception e)` one below.

Reply:
It looks like track 1 only has `catch (RuntimeException e)`. Is there a reason why we needed to broaden this?